Sources/Core/Contract/ContractProtocol.swift

```diff
@@ -263,7 +263,7 @@ extension DefaultContractProtocol {
         let method = Data.fromHex(method) == nil ? method : method.addHexPrefix().lowercased()
 
         // MARK: - Encoding ABI Data flow
-        guard let abiMethod = methods[method]?.first,
+        guard let abiMethod = methods[method]?.first(where: { $0.inputs.count == parameters.count }),
```

Comment:
I think this won't work for methods where there is multiple overloads but with the same parameters amount. As far as I understand this would work in solidity: https://docs.soliditylang.org/en/v0.4.21/contracts.html#overload-resolution-and-argument-matching

Reply:
So per your example if you pass in `f` as a function name you will get two result and the first one will be picked while you probably expect the second one (let's think that just for the sake of example).

But if you specify the full name - you will get 0 or 1 results in the returned array. 
If method ID is passed (which is 4 bytes of sha3 keccak256 of function name and it's parameters) you will get 0 or more results as some clashes can still happen but most of the time it should be 0 or 1 because it's somewhat rare to have first 4 bytes of two hashes of two different functions in the same smart contract to clash.